src/client.ts

```diff
@@ -32,23 +32,36 @@ export default class PloneClient {
     return token;
   };
 
-  loginQuery = (loginArgs: Omit<LoginArgs, 'config'>) => {
-    return _loginQuery({ ...loginArgs, config: this.config });
+  getConfig = () => {
+    return this.config;
   };
+  /*
+    Conventionally, get<Entity>Query naming scheme should be used for
+    objects that are supposed to be used with `useQuery` by the user.
 
-  getContentQuery = (contentArgs: Omit<ContentArgs, 'config'>) => {
-    return _getContentQuery({ ...contentArgs, config: this.config });
-  };
+    Similarily, create<Entity>Mutation naming scheme would be used for
+    objects that are supposed to be used with `useMutation` by the user.
+  */
 
-  createContentQuery = () => {
-    return _createContentQuery({ config: this.config });
-  };
+  /*
+    Initialization queries
+  */
+  loginQuery = queryWithConfig(_loginQuery, this.getConfig);
 
-  updateContentQuery = () => {
-    return _updateContentQuery({ config: this.config });
-  };
-
-  deleteContentQuery = () => {
-    return _deleteContentQuery({ config: this.config });
-  };
+  /*
+    Content queries
+  */
+  getContentQuery = queryWithConfig(_getContentQuery, this.getConfig);
+  createContentMutation = mutationWithConfig(
```

Comment:
Renamed the mutation query objects to `create<Entity>Mutation` pattern. Otherwise the user has no indication about whether they should use `useQuery` or `useMutation` with the object.